src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -0,0 +1,78 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Dapr.Client;
+
+namespace Microsoft.Extensions.Configuration.DaprSecretStore
+{
+    /// <summary>
+    /// A Dapr Secret Store based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    internal class DaprSecretStoreConfigurationProvider : ConfigurationProvider
+    {
+        private readonly string _store;
+
+        private readonly IEnumerable<DaprSecretDescriptor> _secrets;
+
+        private readonly DaprClient _client;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="store">Dapr Secre Store name.</param>
+        /// <param name="secrets">The secrets to retrieve.</param>
+        /// <param name="client">Dapr client used to retrieve Secrets</param>
+        public DaprSecretStoreConfigurationProvider(string store, IEnumerable<DaprSecretDescriptor> secrets, DaprClient client)
+        {
+            if (store == null)
+            {
+                throw new ArgumentNullException(nameof(store));
+            }
+
+            if (secrets == null)
+            {
+                throw new ArgumentNullException(nameof(secrets));
+            }
+
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            _store = store;
+            _secrets = secrets;
+            _client = client;
+        }
+
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task LoadAsync()
+        {
+            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var secret in _secrets)
+            {
+                var result = await _client.GetSecretAsync(_store, secret.SecretName, secret.Metadata).ConfigureAwait(false);
+
+                foreach (var returnedKey in result.Keys)
+                {
+                    var key = returnedKey != "_default" ? returnedKey : secret.SecretName;
```

Comment:
What is "_default"?  Is this for testing?

Reply:
Some secret stores can return many values for a given key (Secret name) and that is the reason the API returns a dictionary. For instance in kubernetes you can define a secret :

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
type: Opaque
data:
  username: YWRtaW4=
  password: MWYyZDFlMmU2N2Rm
```
In this case the secret name or key is `mysecret` and returns two values in the dictionary with `username` and `password` as keys.

For an Azure Key Vault which maps one secret name with one value the API returns one value in the dictionary with `_default` as key

Reply:
Thanks for the explanation.  

    with one value the API return

Can you clarify which API are you referring to here?  When I searched the web "keyvault _default" and the keyvault dotnet sdk I found no references, and the components-contrib repo doesn't have references to "_default".

Reply:
This was recently fixed here: https://github.com/dapr/components-contrib/pull/267/files so this check is no longer needed! I'll change it.

Reply:
Check 481f154